chore: Fix bundle size job

chloeYue opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR is to resolve the bundle size job fails on main branch: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/128879/workflows/4a68c135-363f-4867-bdbe-60cef830b57e/jobs/4615238

The bundle-size check job doesn't really run on PR branch, so we didn't catch it on PR, and they failed directly on main branch:
https://github.com/MetaMask/metamask-extension/blob/main/.circleci/scripts/bundle-stats-commit.sh#L19 


When i use ssh to connect and debug, i find the issue is that Json format is not correct as we are inserting value with echo:

![Screenshot 2025-03-06 at 14 03 41](https://github.com/user-attachments/assets/eebef28f-998e-40da-8e1d-4a06ca9b26b1)

The solution is to use jq to correctly parse json

In addition, this PR also address the following flaws found by @HowardBraham 

* `bundle_size_data.js` and `bundle_size_data.temp.js` should be removed from the repo and added to .gitignore => i updated the spec to only commit the file `bundle_size_data.json`, and i added other 2 files in gitignore
* The overnight scheduled runs on main are causing multiple commits per SHA. If no new SHA is added, no new commit should be made. => i updated the spec to check the SHA, if SHA already exist in json file,  then we exits without making a new commit

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/30817?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

chloeYue committed:
test

chloeYue committed:
fix

chloeYue committed:
revert to only run on main

chloeYue committed:
upgrade trezor connect

chloeYue committed:
upgrade trezor-connect

chloeYue committed:
fix yarn lock

chloeYue committed:
fix dependency

chloeYue committed:
dix depencency

chloeYue commented:
@metamaskbot update-policies

mikesposito committed:
chore: recreate policies (#30835)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/30835?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

chloeYue commented:
@SocketSecurity ignore-all

davidmurdoch committed:
try this

davidmurdoch commented:
@metamaskbot update-policies

metamaskbot committed:
Update LavaMoat policies

seaona committed:
policies update

seaona committed:
dedupe + new policies

seaona committed:
fix

seaona committed:
Frederik's fix

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

chloeYue committed:
Merge branch 'trezor-audit' into chloe-bundle-size

chloeYue committed:
revert changes in policy

chloeYue committed:
Merge branch 'main' into chloe-bundle-size

chloeYue committed:
test on pr branch

chloeYue committed:
Merge branch 'chloe-bundle-size' of https://github.com/MetaMask/metamask-extension into chloe-bundle-size

chloeYue committed:
fix

chloeYue committed:
revert changes to only run on main

HowardBraham commented:
@chloeYue I looked into this job a little more deeply because I was moving it to GitHub Actions.

In addition to the flaw you fixed in this PR, it has the following flaws:
- `bundle_size_data.js` and `bundle_size_data.temp.js` should be removed from the repo and added to `.gitignore`
- The overnight scheduled runs on `main` are causing multiple commits per SHA.  If no new SHA is added, no new commit should be made.

chloeYue committed:
address comment

chloeYue committed:
revert to only run on main branch

chloeYue committed:
Merge branch 'main' into chloe-bundle-size

chloeYue committed:
Merge branch 'main' into chloe-bundle-size

chloeYue committed:
Merge branch 'main' into chloe-bundle-size

chloeYue commented:
> @chloeYue I looked into this job a little more deeply because I was moving it to GitHub Actions.
> 
> In addition to the flaw you fixed in this PR, it has the following flaws:
> 
> * `bundle_size_data.js` and `bundle_size_data.temp.js` should be removed from the repo and added to `.gitignore`
> * The overnight scheduled runs on `main` are causing multiple commits per SHA.  If no new SHA is added, no new commit should be made.

Hi @HowardBraham , thanks for your comment, i updated the spec to only commit the file `bundle_size_data.json` and i also added a check to be sure that we don't commit repeating SHA

seaona commented on .circleci/scripts/bundle-stats-commit.sh:
```diff
@@ -40,41 +40,45 @@ git config --global user.name "MetaMask Bot"
 
-git add .
+# Only add the JSON file
+git add stats/bundle_size_data.json
```
mmm I'm a bit unsure about this change, because at the moment, we are also using the `bundle-size-data.js` file

https://github.com/MetaMask/extension_bundlesize_stats/tree/main/stats

in the visualization tool, so not uploading it will cause that to be updated, no?

https://github.com/MetaMask/extension_bundlesize_stats/blob/main/background.html#L136

chloeYue replied:
I think Howard means we don't need the file `bundle_size_data.js` as it should have same data with  `bundle_size_data.json` , so we can only refer to `bundle_size_data.json`.
 But yeah what you pointed out makes sense, maybe we should first go into [extension_bundlesize_stats](https://github.com/MetaMask/extension_bundlesize_stats) to remove file `bundle_size_data.js` and `bundle-size-data-temp.js`, and also change the line you linked to refer to `bundle_size_data.json`.
 Converting this PR to draft, and tag @davidmurdoch for clarifications.

HowardBraham replied:
@seaona @chloeYue I made the changes to [extension_bundlesize_stats](https://github.com/MetaMask/extension_bundlesize_stats), and this PR should be good to go now

chloeYue replied:
Thanks a lot @HowardBraham

seaona replied:
awesome , thank you so much both!

HowardBraham committed:
Merge branch 'main' into chloe-bundle-size

chloeYue committed:
Merge branch 'main' into chloe-bundle-size

chloeYue committed:
test

chloeYue committed:
revert change for testing

davidmurdoch reviewed (dismissed):
approved, with some optional suggestions if you want.

chloeYue committed:
remove unnecessary steps

chloeYue commented on .circleci/scripts/bundle-stats-commit.sh:
```diff
@@ -40,41 +40,45 @@ git config --global user.name "MetaMask Bot"
 
 cp temp/stats/bundle_size_data.temp.js temp/stats/bundle_size_data.js
-
 echo " }" >> temp/stats/bundle_size_data.js
```
make sense ! 
i updated to remove unnecessary steps in this commit: https://github.com/MetaMask/metamask-extension/pull/30817/commits/21ac077feaa04096772e13a12efce554f25e4831

HowardBraham commented on .circleci/scripts/bundle-stats-commit.sh:
```diff
@@ -40,41 +40,45 @@ git config --global user.name "MetaMask Bot"
 
 git clone git@github.com:MetaMask/extension_bundlesize_stats.git temp
```
Yeah you're right, it's the "correct" thing to do.  As for actual impact, it shaves off a few seconds from one job that only runs on `main`.  But I guess there's no disadvantage?

chloeYue replied:
thanks both, i think it makes sense, i only need to clone the latest commit.
i addressed in the commit: https://github.com/MetaMask/metamask-extension/pull/30817/commits/e234f9eb410a7a0550d8a3a38930a0acd4ce745a

chloeYue committed:
set clone depth to one
